src/api/api.ts

```diff
@@ -124,6 +146,85 @@ class API {
     }
   }
 
+  async getChatRoomWithGivenUser(
+    participantEmail: string,
+    acceptTimezone: string | null
+  ): Promise<ChatRoom> {
+    try {
+      const headers: HeadersInit = {
+        "Accept-Role": store.user.profile!.selected_profile!.toUpperCase(),
```

Comment:
`.toUpperCase()` jest tutaj zbędny bo `selected_profile` ma typ taki, że może przyjąć wartości tylko `CARETAKER` lub `CLIENT`, które już sa upperCasem:  
```suggestion
        "Accept-Role": store.user.profile!.selected_profile,
```

Reply:
Fakt